yt_dlp/extractor/iqiyi.py

```diff
@@ -392,3 +405,269 @@ def _real_extract(self, url):
+                    var enc_params = [];
+                    for (var prop in query) {
+                        enc_params.push(encodeURIComponent(prop) + '=' + encodeURIComponent(query[prop]));
+                    }
+                    var dash_path = '/dash?' + enc_params.join('&'); dash_path += '&vf=' + cmd5x(dash_path);
+                    dash_paths[bid] = dash_path;
+                });
+                return JSON.stringify(dash_paths);
+            }));
+            saveAndExit();
+        """ % (tvid, vid, dfp, mode, lang, js_bid_list, cmd5x_func))[1].strip(), video_id)
+
+        formats, subtitles = [], {}
+        initial_format_data = self._download_json(
+            urljoin('https://cache-video.iq.com', dash_paths['0']), video_id,
+            note='Downloading initial video format info', errnote='Unable to download initial video format info')['data']
+
+        preview_time = traverse_obj(initial_format_data, ('boss_ts', 'data', 'previewTime'), expected_type=float_or_none)
+        if preview_time:
+            self.report_warning(f'This preview video is limited to {preview_time} seconds')
+
+        # TODO: Extract audio-only formats
+        for bid in set(traverse_obj(initial_format_data, ('program', 'video', ..., 'bid'), expected_type=str_or_none)):
+            dash_path = dash_paths.get(bid)
+            if not dash_path:
+                self.report_warning(f'Unknown format id: {bid}. It is currently not being extracted')
+                continue
+            format_data = traverse_obj(self._download_json(
+                urljoin('https://cache-video.iq.com', dash_path), video_id,
+                note=f'Downloading format data for {self._BID_TAGS[bid]}', errnote='Unable to download format data',
+                fatal=False), 'data', expected_type=dict)
+
+            video_format = next((video_format for video_format in traverse_obj(
+                format_data, ('program', 'video', ...), expected_type=dict) if str(video_format['bid']) == bid), {})
+            extracted_formats = []
+            if video_format.get('m3u8Url'):
+                extracted_formats.extend(self._extract_m3u8_formats(
+                    urljoin(format_data.get('dm3u8', 'https://cache-m.iq.com/dc/dt/'), video_format['m3u8Url']), 'mp4', m3u8_id=bid), fatal=False)
+            if video_format.get('mpdUrl'):
+                # TODO: Properly extract mpd hostname
+                extracted_formats.extend(self._extract_mpd_formats(
+                    urljoin(format_data.get('dm3u8', 'https://cache-m.iq.com/dc/dt/'), video_format['mpdUrl']), mpd_id=bid), fatal=False)
+            if video_format.get('m3u8'):
+                ff = video_format.get('ff', 'ts')
+                if ff == 'ts':
+                    m3u8_formats, _ = self._parse_m3u8_formats_and_subtitles(
+                        video_format['m3u8'], 'data:application/x-mpegurl;base64,' + base64.b64encode(video_format['m3u8'].encode('utf-8')).decode('ascii'),
```

Comment:
Any better way to do this? data: URLs spam the verbose log

Reply:
try this:
```diff
diff --git a/yt_dlp/YoutubeDL.py b/yt_dlp/YoutubeDL.py
index 71369bc44..5cb5c4e33 100644
--- a/yt_dlp/YoutubeDL.py
+++ b/yt_dlp/YoutubeDL.py
@@ -2748,7 +2748,9 @@ class YoutubeDL(object):
         if not test:
             for ph in self._progress_hooks:
                 fd.add_progress_hook(ph)
-            urls = '", "'.join([f['url'] for f in info.get('requested_formats', [])] or [info['url']])
+            urls = '", "'.join(
+                f['url'].split(',')[0] + '<data>' if f['url'].startswith('data:') else f['url']
+                for f in info.get('requested_formats', []) or info)
             self.write_debug('Invoking downloader on "%s"' % urls)

         # Note: Ideally info should be a deep-copied so that hooks cannot modify it.
```

Reply:
Modified it to be:
```py
            urls = '", "'.join(
                (f['url'].split(',')[0] + ',<data>' if f['url'].startswith('data:') else f['url'])
                for f in info.get('requested_formats', []) or [info])
```
Does this look okay?

Reply:
yes. You can commit it to this PR